Create create.md (#3142)
Codecademy/docs: content/lua/concepts/tables/terms/create/create.md

---
title: 'create()'
description: 'Returns a new table with the provided value repeated the provided amount of times.'
Subjects:
  - 'Computer Science'
  - 'Game Development' 
Tags:
  - 'Tables'
  - 'Functions'
CatalogContent:
  - 'learn-lua'
 - 'paths/computer-science'
---

> The **.create()** function in Lua Tables generates a new table with a specified value repeated a given number of times.

Reviewer: ```suggestion
The **`.create()`** function generates a new table with a specified value repeated a given number of times.
```

## Syntax

```lua
table.create(value, amount)
```

> The `create` function is used to generate a new table with a specified value repeated a certain number of times. This function is helpful when you need to initialize a table with a repetitive pattern of values.

Reviewer: Instead of repeating the intro sentence, please describe the syntax example, explaining what `value` and `amount` are.

## Example

Consider the following table:

> ```lua

Reviewer: Can we have the example print out the values in the table so the following ` ```shell` block can actually be output from the example.

local newTable = table.create("Hello", 3)
```

After executing this code, the newTable will contain the following elements:

> ```shell

Reviewer: We prefer this to be output from the example code. Can you revise the example so an output block would make sense?

---

> title: 'create()'

Reviewer 2: ```suggestion
Title: '.create()'
```

> description: 'Returns a new table with the provided value repeated the provided amount of times.'

Reviewer 2: ```suggestion
Description: 'Returns a new table with a specified value repeated a given number of times.'
```

Subjects:
  - 'Computer Science'
  - 'Game Development' 
Tags:
  - 'Tables'
  - 'Functions'
CatalogContent:
  - 'learn-lua'

>  - 'paths/computer-science'

Reviewer 2: ```suggestion
  - 'paths/computer-science'
```

---

The **`.create()`** function generates a new table with a specified value repeated a given number of times.

## Syntax

> ```lua

Reviewer 2: ```suggestion
```pseudo
```

> table.create(value, amount)

Author: 🫠🫠

```

- `value` : This parameter represents the value to populate the table with.
- `amount` : This parameter specifies how many times to repeat the `value` in the table.

> 

Reviewer 2: Let's add a note here explaining what happens when `value` is nill or is not given

```suggestion

> **Note:** When `value` is omitted or `nil`, the table will be returned empty but will have preallocated space for the `amount` specified. Preallocation is done on the array portion of the table, making `table.create` counter-productive on dictionaries.

```

---
title: 'create()'
description: 'Returns a new table with the provided value repeated the provided amount of times.'
Subjects:
  - 'Computer Science'
  - 'Game Development' 
Tags:
  - 'Tables'
  - 'Functions'
CatalogContent:
  - 'learn-lua'
 - 'paths/computer-science'
---

The **`.create()`** function generates a new table with a specified value repeated a given number of times.

## Syntax

```lua
table.create(value, amount)
```

> **value** : This parameter represents the value you want to populate the table with.
> 
> **amount** : This parameter specifies how many times you want to repeat the **value** in the table.

Reviewer: ```suggestion
- `value` : This parameter represents the value to populate the table with.
- `amount` : This parameter specifies how many times to repeat the `value` in the table.
```
We avoid first- and second-person pronouns in Docs.

## Example

Consider the following table:

```lua
local newTable = table.create("Hello", 3)
console.log(newTable)
```

> 
> 
> After executing this code, the newTable will contain the following elements:

Reviewer: ```suggestion

After executing this code, it will produce the following output:
```
